src/main/java/frc/robot/commands/ToFIntakeCommand.java

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.commands;
+
+import frc.robot.subsystems.ArmSubsystem;
+import frc.robot.subsystems.IntakeSubsystem;
+import frc.robot.subsystems.ArmSubsystem.ArmState;
+import frc.robot.subsystems.IntakeSubsystem.FlightStates;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+
+public class ToFIntakeCommand extends CommandBase {
+  /** Creates a new ToFIntakeCommand. */
+  private final IntakeSubsystem intakeSubsystem;
+  private final ArmSubsystem armSubsystem;
+
+  public ToFIntakeCommand(IntakeSubsystem intakeSubsystem, ArmSubsystem armSubsystem) {
+    this.intakeSubsystem = intakeSubsystem;
+    this.armSubsystem = armSubsystem;
+    addRequirements(intakeSubsystem);
+    addRequirements(armSubsystem);
+  }
+
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize() {} 
+
+  // Called every time the scheduler runs while the command is scheduled.
+  @Override
+  public void execute() {
+    FlightStates tofState = intakeSubsystem.getFlightState();
+
+    switch (tofState) {
+      case IDLE:
+        if (intakeSubsystem.getSpeed() != 0) {
+          intakeSubsystem.stop();
+        }
+        break;
+
+      case CONE:
+        // If we're about to shoot, make sure not to stay running intake normally
+        if (armSubsystem.getArmState().equals(ArmState.HIGHDROP)) {
+          tofState = FlightStates.CONE_SCORE;
```

Comment:
I don't expect this will work because the tofState value will be overwritten by getFlightState() before the switch is reached again.

Reply:
Could this be fixed by adding an additional check in changeFlightState() to only switch to cone mode if we're not already in cone score mode?

Reply:
Hopefully should be fixed with new case/switch structure